### PR TITLE
Lib: support selection by name

### PR DESCRIPTION
**Pull request by moritzraho**
## Description
Feature for https://github.com/adobe/aio-cli-plugin-console/pull/157

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by moritzraho** [comment]
cc @Himavanth @sandeep-paliwal  @purplecabbage @rajarju  @shazron @meryllblanchet

**Comment by purplecabbage** [comment]
I  ❤️    the idea here, but why not just change the implementation in findOrgOrThrow?
It could just be:
```
function findOrgOrThrow (orgIdOrName, organizations) {
  const selectedOrg = organizations
    .find(org => org.id === orgIdOrName || org.name === orgIdOrName)
  if (!selectedOrg) {
    throw new Error(`Org ${orgIdOrName} not found`)
  }
  return selectedOrg
}

function findProjectOrThrow (projectIdOrName, projects) {
  const selectedProject = projects
    .find(p => p.id === projectIdOrName || p.name === projectIdOrName)
  if (!selectedProject) {
    throw new Error(`Project ${projectIdOrName} not found`)
  }
  return selectedProject
}

function findWorkspaceOrThrow (workspaceIdOrName, workspaces) {
  const selectedWorkspace = workspaces
    .find(p => p.id === workspaceIdOrName || p.name === workspaceIdOrName)
  if (!selectedWorkspace) {
    throw new Error(`Workspace ${workspaceIdOrName} not found`)
  }
  return selectedWorkspace
}
```

**Comment by moritzraho** [comment]
@purplecabbage good point I initially wanted to do that, but then I am trying to avoid a breaking change for so little, as the `./lib/console-cli.js` file is required by `aio-cli-plugin-app`, `aio-cli-plugin-consolee` and by `./generators/app` (init)

File: lib/console-cli.js [hunk 1]
```diff
@@ -402,14 +393,16 @@ class LibConsoleCLI {
     return helpers.findFirstEntpCredential(credentials)
   }
 
-  async getWorkspaceConfig (orgId, projectId, workspaceId, supportedServices) {
+  async getWorkspaceConfig (orgId, projectId, workspaceId, supportedServices = null) {
     spinner.start('Downloading Workspace config...')
-    const json = (await this.sdkClient.downloadWorkspaceJson(orgId, projectId, workspaceId)).body
+    let json = (await this.sdkClient.downloadWorkspaceJson(orgId, projectId, workspaceId)).body
```
Comment (rajarju) [inline lib/console-cli.js:398]:
```suggestion
 Fancy suggestion :D 
    let { body: json } = await this.sdkClient.downloadWorkspaceJson(orgId, projectId, workspaceId)
```

**Review by rajarju — APPROVED** [review]

**Pull request merged**